redis/server.hh

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2019 pengjian.uestc @ gmail.com
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "auth/service.hh"
+#include "service/storage_proxy.hh"
+#include "service_permit.hh"
+#include "redis/query_processor.hh"
+#include "redis/request.hh"
+#include "cql3/values.hh"
+#include "auth/authenticator.hh"
+#include "timeout_config.hh"
+#include <memory>
+#include <seastar/net/tls.hh>
+#include <seastar/core/semaphore.hh>
+#include "seastar/core/distributed.hh"
+#include "seastar/core/reactor.hh"
+#include <seastar/core/metrics_registration.hh>
+#include "utils/fragmented_temporary_buffer.hh"
+#include "utils/estimated_histogram.hh"
+#include "redis/protocol_parser.hh"
+#include "redis/request.hh"
+#include "redis/reply.hh"
+#include "redis/options.hh"
+#include  "service/client_state.hh"
+
+db::consistency_level make_consistency_level(const sstring&);
+
+namespace db {
+class config;
+};
+class redis_exception;
+
+namespace redis_transport {
+
+struct redis_server_config {
+    ::timeout_config _timeout_config;
+    size_t _max_request_size;
+    db::consistency_level _read_consistency_level;
+    db::consistency_level _write_consistency_level;
+    size_t _total_redis_db_count;
+};
+
+class redis_server {
+    std::vector<server_socket> _listeners;
+    distributed<service::storage_proxy>& _proxy;
+    distributed<redis::query_processor>& _query_processor;
+    redis_server_config _config;
+    size_t _max_request_size;
+    semaphore _memory_available;
+    seastar::metrics::metric_groups _metrics;
+private:
+    uint64_t _connects = 0;
+    uint64_t _connections = 0;
```

Comment:
Is it repeated with _total_connections & _current_connections in line 145?

```
    uint64_t _total_connections = 0;
    uint64_t _current_connections = 0;
```

Reply:
BTW, it's difficult to know the difference between those two variables from the name.

Reply:
total_connections, means that this is an incremental counter, while current_connections means that current active client connections counter which increased when connection is connected, reduced when the connection is closed.